platform/components/docker-gbsyncd-credo/critical_processes.j2

```diff
@@ -0,0 +1 @@
+program:syncd
```

Comment:
I noticed that docker-gbsyncd-vs is using gbsyncdmgrd script from sonic-sairedis repo which checks if gearbox_config.json exists or not, if not exist then it will not start the syncd process.

Reply:
one question here is that how do we know the gearbox is available on this platform? for example, many broadcom platformd does not have gearbox, and therefore, there is no need to start gbsyncd docker. gbsyncdmgrd is one way to handle such issue, but a better solution might be not start the gbsyncd docker at all if gearbox is not available on this platform.

Reply:
Currently, it is handled by platform/components/docker-gbsyncd-credo/docker-init.sh, for which the container exits by itself if no gearbox_config.json.

I'll check whether it can be changed to gbsyncdmgrd or not starting the container at all.

In platform/components/docker-gbsyncd-credo/docker-init.sh,
GB_CONFIG=/usr/share/sonic/hwsku/gearbox_config.json

if [ ! -f $GB_CONFIG ]; then
   exit 0
fi

Reply:
To address the problem of not starting gbsyncd containers unrelated to a platform, the change "Separate gbsyncd-credo service" is committed. 

After the change, it is assumed that each gbsyncd variant has its own service and docker image. For example, for gbsyncd-credo, there are gbsyncd-credo.service and gbsyncd-credo docker; for gbsyncd-vs, there are gbsyncd-vs.service and gbsyncd-vs docker. Potentially, a new service file and docker image should be created if there is a new type of gbsyncd added in the future.

In the booting, all kinds of gbsyncd services installed will all be started, and each of them use ExecCondition=/usr/bin/gbsyncd-platform.sh to determine whether to start the corresponding docker or just skip it, by checking the file gbsyncd.ini in the device platform folder. If the result is skipping, it will leave some messages in the log, e.g.,

Jul 22 21:49:06.264558 gd270 INFO systemd[1]: Starting gbsyncd-credo service...
Jul 22 21:49:07.117152 gd270 NOTICE systemd[1]: gbsyncd-credo.service: Control process exited, code=exited, status=1/FAILURE
Jul 22 21:49:07.117328 gd270 INFO systemd[1]: gbsyncd-credo.service: Skipped due to 'exec-condition'.
Jul 22 21:49:07.118970 gd270 INFO systemd[1]: Condition check resulted in gbsyncd-credo service being skipped.